xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIMEGLES.cpp

```diff
@@ -43,15 +44,32 @@ bool CRendererDRMPRIMEGLES::Configure(const VideoPicture &picture, float fps, un
   for (auto &texture : m_DRMPRIMETextures)
     texture.Init(winSystem->GetEGLDisplay());
 
+  for (auto& fence : m_fences)
+  {
+    fence.reset(new CEGLFence(winSystem->GetEGLDisplay()));
+  }
+
   return CLinuxRendererGLES::Configure(picture, fps, orientation);
 }
 
 void CRendererDRMPRIMEGLES::ReleaseBuffer(int index)
 {
+  m_fences[index]->DestroyFence();
+
   m_DRMPRIMETextures[index].Unmap();
   CLinuxRendererGLES::ReleaseBuffer(index);
 }
 
+bool CRendererDRMPRIMEGLES::NeedBuffer(int index)
+{
+  if (m_fences[index]->IsSignaled())
```

Comment:
`return !m_fences[index]->IsSignaled()`?

Reply:
If you think it's more clear then sure I can do that ❤️

Reply:
I just like more concise code personally, it is not "wrong" as-is by any means.

I don't always write the disclaimer, but I think you know that most of the time I'm just making personal suggestions for alternatives in which you could argue for either side :) Here I think the argument would be that, written as you did it, it is easier to extend the function with other checks should the need arise.